src/core/tUnification.ml

```diff
@@ -992,6 +993,10 @@ and unify_with_variance uctx f t1 t2 =
 		List.iter2 unify_nested tl1 tl2
 	| TAbstract(a1,tl1),TAbstract(a2,tl2) when a1 == a2 ->
 		List.iter2 unify_nested tl1 tl2
+	| TAbstract(ab,tl),_ when Meta.has Meta.ForwardVariance ab.a_meta ->
+		unify_with_variance uctx f (get_this_type ab tl) t2
+	| _,TAbstract(ab,tl) when Meta.has Meta.ForwardVariance ab.a_meta ->
+		unify_with_variance uctx f t1 (get_this_type ab tl)
```

Comment:
That second `t1` should be `t2`.

Reply:
... which also suggests that we need an expected-failure test for this case.

Reply:
Are you talking about `tl` this is type list, not type one. Or did i misunderstood?

Reply:
Oh wow, I need new glasses... never mind!